pkg/services/dashboards/accesscontrol.go

```diff
@@ -49,7 +49,9 @@ func NewFolderNameScopeResolver(folderDB folder.FolderStore, folderSvc folder.Se
 		if len(nsName) == 0 {
 			return nil, ac.ErrInvalidScope
 		}
-		folder, err := folderDB.GetFolderByTitle(ctx, orgID, nsName)
+		// this will fetch only root folders
+		// this is legacy code so most probably it is not used
+		folder, err := folderDB.GetFolderByTitle(ctx, orgID, nsName, nil)
```

Comment:
@grafana/identity-access-team is this resolver used for newly created folders?

Reply:
Do we then need to update the checks [here](https://github.com/grafana/grafana/blob/main/pkg/services/ngalert/api/authorization.go#L22-L30) to use a UID based scope instead of a name based scope?

So have 
```
eval = ac.EvalPermission(ac.ActionAlertingRuleDelete, dashboards.ScopeFoldersProvider.GetResourceScopeUID(ac.Parameter(":Namespace")))
```
instead of
```
eval = ac.EvalPermission(ac.ActionAlertingRuleDelete, dashboards.ScopeFoldersProvider.GetResourceScopeName(ac.Parameter(":Namespace")))
```

Reply:
good catch!